sdk/synapse/Azure.Analytics.Synapse.Artifacts/tests/NotebookClientLiveTests.cs

```diff
@@ -50,23 +50,32 @@ public async Task TestCreateNotebook()
                 nbformatMinor: 2,
                 new List<NotebookCell>()
             );
-            var operation = await NotebookClient.StartCreateOrUpdateNotebookAsync("MyNotebook", new NotebookResource(notebook));
-            while (!operation.HasValue)
-            {
-                operation.UpdateStatus();
-            }
-            Assert.AreEqual("MyNotebook", operation.Value.Name);
+            string notebookName = Recording.GenerateName("Notebook");
+            NotebookCreateOrUpdateNotebookOperation operation = await NotebookClient.StartCreateOrUpdateNotebookAsync(notebookName, new NotebookResource(notebook));
+            NotebookResource notebookResource = operation.WaitForCompletionAsync().ConfigureAwait(true).GetAwaiter().GetResult();
+            Assert.AreEqual(notebookName, notebookResource.Name);
         }
 
         [Test]
         public async Task TestDeleteNotebook()
         {
-            var operation = await NotebookClient.StartDeleteNotebookAsync("MyNotebook");
-            while (!operation.HasValue)
-            {
-                operation.UpdateStatus();
-            }
-            Assert.AreEqual(200, operation.Value.Status);
+            string notebookName = Recording.GenerateName("Notebook");
+
+            Notebook notebook = new Notebook(
+                new NotebookMetadata
+                {
+                    LanguageInfo = new NotebookLanguageInfo(name: "Python")
+                },
+                nbformat: 4,
+                nbformatMinor: 2,
+                new List<NotebookCell>()
```

Comment:
All collections should default to an empty collection anyway. If this is part of the public API, this should not be necessary.

If this is an internal constructor used for deserialization, you should not be using it in tests if avoidable. It can change slightly. It's important to test the public APIs to make sure 1) they work, and 2) behavior doesn't unexpectedly changed (especially after GA).

Reply:
Thanks so much for reviewing my code so carefully. These parameters are needed by the constructor. And for the collection, it should not be null base on the code. Here is the link: [Notebook.cs](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/Notebook.cs#L31)